### PR TITLE
Fix error handling in read() / download()

### LINKED ISSUE #1184: More informative error reporting upon opening corrupted data products
#### Problem description

Right now, when `Lightkurve` opens a FITS file that is corrupted, it produces some cryptic low-level error
(usually suggesting that the file is truncated).

Such cases occur in practice possibly due to download error, e.g., issue #1173 . I recalled it was the root cause of an old issue in using `CBVCorrector` (where it tries to query and download neighboring lightcurves, a lightcurve file is corrupted).

#### Suggestion

Improve error handling (in reading lc / tpf files) so that
1. it explicitly informs users that the file is possibly corrupted.
2. include the actual path of the offending file in the error message

#### Environment

-  lightkurve version (e.g. 1.0b6): 2.1.1

### LINKED ISSUE #1211: Download fits files from MAST: raises misleading FileNotFoundError when MAST returns an HTTP error
#### Problem description

Currently, when one users `Lightkurve` to download LC / TPF files from MAST, and MAST returns HTTP error, `Lightkurve` would incorrectly proceed as if the download is successful, and ultimately raises a FileNotFoundError.

#### Example

```python
import lightkurve as lk

tic, sector = 424391564, 50
# MAST happens to return an HTTP 502 error when the codes try to download the FITS. See below for more.
lc = lk.search_lightcurve(f"TIC{tic}", sector=sector)[0].download()
```

```python
File C:\dev\lightkurve\src\lightkurve\utils.py:555, in suppress_stdout.<locals>.wrapper(*args, **kwargs)
    553 sys.stdout = devnull
    554 try:
--> 555     return f(*args, **kwargs)
    556 # restore to default
    557 finally:
    558     sys.stdout = old_out

File C:\dev\lightkurve\src\lightkurve\search.py:529, in SearchResult.download_all(self, quality_bitmask, download_dir, cutout_size, **kwargs)
    526 products = []
    527 for idx in range(len(self.table)):
    528     products.append(
--> 529         self._download_one(
    530             table=self.table[idx : idx + 1],
......
File C:\pkg\_winNonPortables\Anaconda3\envs\lkv2_1_dev\lib\site-packages\astropy\io\fits\util.py:388, in fileobj_open(filename, mode)
    378 def fileobj_open(filename, mode):
    379     """
    380     A wrapper around the `open()` builtin.
    381 
   (...)
    385     a raw random-access file reader.
    386     """
--> 388     return open(filename, mode, buffering=0)
FileNotFoundError: [Errno 2] No such file or directory: 'data/mastDownload\\TESS\\tess2022085151738-s0050-0000000424391564-0222-s\\tess2022085151738-s0050-0000000424391564-0222-s_lc.fits'
```

#### Expected behavior

`Lightkurve` should report the actual MAST HTTP error, and ensure there is a way for users to recover.

#### Environment

-  lightkurve version (e.g. 1.0b6): 2.2

### PR DESCRIPTION
**Pull request by orionlee**
Close #1184 : more informative error when reading a truncated FITS file
Close #1211 : raise proper error when MAST returns HTTP error

### CODE REVIEW

**Comment by orionlee** [comment]
For reference: The unit test for truncated FITS file that does not work on Windows (referenced in the test comment) due to `PermissionError` in temporary directory cleanup. (It is replaced by a lower-level test in `test_read.py`)
```python
import shutil
from . import TESTDATA

@pytest.mark.filterwarnings("ignore:.*been truncated.*")  # ignore AstropyUserWarning: File may have been truncated
def test_corrupt_download_handling_case_truncated():
    """Regression test for #1184; ensure download() yields an error that includes the filename."""
    with tempfile.TemporaryDirectory() as tmpdirname:
        # Pretend a corrupt file exists at the expected cache location
        expected_dir = os.path.join(
            tmpdirname, "mastDownload", "TESS", "tess2018206045859-s0001-0000000261136679-0120-s"
        )
        expected_fn = os.path.join(
            expected_dir, "tess2018206045859-s0001-0000000261136679-0120-s_lc.fits"
        )
        os.makedirs(expected_dir)
        filename_lc_pimen_corrupted_src = os.path.join(TESTDATA, "test-lc-tess-pimen-100-cadences-corrupted.fits")
        shutil.copy(filename_lc_pimen_corrupted_src, expected_fn)  # copy a truncated FITS file to cache
        with pytest.raises(LightkurveError) as err:
            search_lightcurve("TIC261136679", mission="TESS", author="SPOC", sector=1, cadence="short").download(
                download_dir=tmpdirname
            )
        assert "may be corrupt" in err.value.args[0]
        assert expected_fn in err.value.args[0]

```

**Comment by orionlee** [comment]
Ready for review.

Even if it's ready to be merged, I intend to hold off the merge until PR #1217 is merged, to avoid creating complication due to conflicting changes.

File: tests/test_search.py [hunk 1]
```diff
@@ -367,6 +374,35 @@ def test_corrupt_download_handling():
...
+        """Mock the `download_product()` response to simulate MAST returns HTTP error"""
+        print("DBG mock_http_error_response called")
+        return Table(data={
+            "Local Path": ["./mastDownload/acme_lc.fits"],
+            "Status": ["ERROR"],
+            "Message": ["HTTP Error 500: Internal Server Error"],
+            "URL": [remote_url],
+            })
+
+    monkeypatch.setattr(Observations, "download_products", mock_http_error_response)
```
Comment (barentsen) [inline tests/test_search.py:398]: I'm loving the use of `monkeypatch` here!

**Comment by barentsen** [comment]
I think this is good to merge once #1217 is merged! :+1:

**Comment by orionlee** [comment]
I intend to merge this PR by Wednesday July 6 (after #1217). Reply back if there is any concern.

**Pull request merged**